### PR TITLE
Add fapolicy rules to allow execution of generated java code

### PR DESCRIPTION
**Pull request by fmarco76**
When fapolicy daemon is running the default rules deny the execution of java code from not trusted sources. This deny also the execution of some code generated inside the instance `work` folder.

This is the case when FIPS is enabled and DISA STIG profile is applied.

To solve the problem a custom rule is applied during the installation.

Solve the  Bug 2091993

### CODE REVIEW

File: base/server/python/pki/server/deployment/scriptlets/fapolicy_setup.py
Comment (edewata) [inline base/server/python/pki/server/deployment/scriptlets/fapolicy_setup.py:40]: This probably should say File Access Policy instead of SELinux.

File: base/server/python/pki/server/deployment/scriptlets/fapolicy_setup.py
Comment (edewata) [inline base/server/python/pki/server/deployment/scriptlets/fapolicy_setup.py:35]: Looks like existing files use dashes (`-`) instead of underscores (`_`), so we probably should follow the convention.

Comment (edewata) [reply base/server/python/pki/server/deployment/scriptlets/fapolicy_setup.py:35]: The example here uses `80-myapps.rules` so maybe we want to use `80` instead of `61`?
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/assembly_blocking-and-allowing-applications-using-fapolicyd_security-hardening#proc_adding-custom-allow-and-deny-rules-for-fapolicyd_assembly_blocking-and-allowing-applications-using-fapolicyd

File: base/server/python/pki/server/deployment/scriptlets/fapolicy_setup.py
Comment (edewata) [inline base/server/python/pki/server/deployment/scriptlets/fapolicy_setup.py:85]: Typo?

**Review by edewata — COMMENTED** [review]
I'm not really familiar with `fapolicyd` but it might be easier & cleaner to create a separate `.rules` file for each PKI instance (e.g. `80-pki-<instance>.rules`). That way we can store more complex rules if necessary and simply delete the file during uninstall, no need to worry about removing specific lines. Also, in case in the future we allow instances to be owned by non-root users, the rules can be owned by different users too.

File: base/server/python/pki/server/deployment/scriptlets/fapolicy_setup.py
Comment (fmarco76) [reply base/server/python/pki/server/deployment/scriptlets/fapolicy_setup.py:35]: > we want to use `80` instead of `61`?

Actually, the rules denying the execution are in the file starting with 70:

`70 - trusted language rules `

therefore we have to use a smaller number because rules are applied in order.

Comment (edewata) [reply base/server/python/pki/server/deployment/scriptlets/fapolicy_setup.py:35]: Sure, that's fine.

File: base/server/python/pki/server/deployment/scriptlets/fapolicy_setup.py
Comment (edewata) [inline base/server/python/pki/server/deployment/scriptlets/fapolicy_setup.py:66]: This is fine for now, but if the rules are becoming more complex we probably should move it into a separate file to make it easier to review/edit the rules, then use it as a template to generate the actual `.rules` file.

**Review by edewata — COMMENTED** [review]
Thanks for the update! Please see my comments.

File: base/server/python/pki/server/deployment/scriptlets/fapolicy_setup.py
Comment (fmarco76) [reply base/server/python/pki/server/deployment/scriptlets/fapolicy_setup.py:66]: Yes, for complex rules a template will be better. However, I think we should avoid this rules changes and find a way to add the generated files into the [fapolicy truststore](https://www.mankier.com/5/fapolicyd.trust) becasue these are already allowed so there are not risks to have conflicts in case other non default rules are applied.

File: base/server/python/pki/server/deployment/scriptlets/fapolicy_setup.py
Comment (edewata) [inline base/server/python/pki/server/deployment/scriptlets/fapolicy_setup.py:76]: I think this check is redundant since now we check the `fapolicy_rule_file`, but I'll leave that to you.

**Review by edewata — APPROVED** [review]
Thanks for the update! I have a minor comment, but it LGTM as is. Feel free to update/merge.

**Comment by fmarco76** [comment]
@edewata Thanks!

**Comment by edewata** [comment]
Oh, I just realized that this PR will create the rules for new instances only. If you want to create the rules for existing instances, you'll probably need to convert the code in the `spawn()` method into an upgrade script. That way when the instance is restarted the rules will be created automatically.

The code in `destroy()` can remain for instance removal.

**Comment by fmarco76** [comment]
If you have an existing instance then you should already have a policy applied. As reference, [the IdM installation documentation](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/installing_identity_management/preparing-the-system-for-ipa-server-installation_installing-identity-management#proc_ensuring-that-fapolicyd-rules-do-not-block-idm-installation_preparing-the-system-for-ipa-server-installation) said to create the rule before the installation. Therefore, I think in case of upgrade could be problematic modify the rules.

**Comment by edewata** [comment]
Do we need to worry about the case where FIPS and DISA STIG are enabled after installation?

**Comment by fmarco76** [comment]
That could be a point but for the tests  I have followed RHEL instruction which explain how to install DISA STIG during OS installation. I do not know if it is possible/automatic after the installation. For FIPS is possible but not recommended.
Nevertheless, in case it is possible to enable DISA STIG after the installation (maybe only fapolicyd), then it would be easier to add the created files into the truststore. It is a single command and do not conflict with the other rules so in any case the approach would be different.

**Pull request merged**
